Lib/traceback.py

```diff
@@ -465,6 +465,9 @@ class TracebackException:
     - :attr:`msg` For syntax errors - the compiler error message.
     """
 
+    def hello(self, exc_value):
```

Comment:
what's this?

Reply:
Per my comment [here](https://github.com/python/cpython/pull/24460#discussion_r571496279), the surprising fact that this seemingly no-op function is _necessary_ for tests to pass is what tells me there's a bug in whatever mechanism is used to clear the `tstate->recursion_headroom` flag.

Solving this seems well out of scope for this PR. (I just pushed this so that you could see what was going on.) I'll remove this no-op `hello` as well as the problematic test case.

Reply:
can you create a standalone script to show what happened, and report that as a separate issue?

I think what happened here was that when format_exception is called in the except block, the recursion limit was already reached, so things go haywire. I don't think it's an exponential blowup in traversing the context and cause links.

Reply:
Maybe I got it wrong - it was in the except after the recursion returned. I haven't tried running it yet, will try to look into it tomorrow.

Reply:
Yes, if you check out `8550dbdd53b87dde1f32288f250507bc43b68130` then without the call to `hello()` there's a fatal error regardless of the indentation of `format_exception()` in the test case.

I tried for a bit to make a minimum example script but wasn't able to replicate. I agree there's no exponential blowup (I was mistaken in my long comment in the other thread) because the `seen` set prevents revisiting nodes in the implicit binary tree I was imagining. Will try some more later.

I still have no explanation as to why the headroom flag isn't getting cleared when we pop out of the recursion error that happens inside the TracebackException constructor (i.e., the second recursion error in the unit test overall in time).